DIP/DIP/Program.cs

```diff
@@ -1,8 +1,14 @@
+using DIP;
+using DIP.Interfaces;
+using DIP.Models;
+
 var builder = WebApplication.CreateBuilder(args);
 
 // Add services to the container.
 
 builder.Services.AddControllers();
+builder.Services.AddScoped(typeof(IPaymentService<>), typeof(CardPaymentService<>));
```

Comment:
why do we need this as scoped?

Reply:
This should be **Transient** Since it's a stateless service and it's lightweight to instantiate and if it's created multiple times in a request it doesn't effect (operational services)
as for scoped this is used when we have a state and we shouldn't create multiple instances in the same request (DBContext)
as for singleton this is one instance for the whole application lifetime (Queue)